src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java

```diff
@@ -683,14 +743,23 @@ private String deleteRequest(String path) throws IOException {
         String path = parent.getPath();
         String ref = Util.rawEncode(parent.getRef());
         int start=0;
-        String url = String.format(API_REPOSITORY_PATH+"/browse/%s?at=%s", owner, repositoryName, path, ref);
-        String response = getRequest(url + String.format("&start=%s&limit=%s", start, 500));
+        UriTemplate template = UriTemplate
+                .fromTemplate(API_BROWSE_PATH + "{&start,limit}")
+                .set("owner", getUserCentricOwner())
+                .set("repo", repositoryName)
+                .set("path", path)
+                .set("at", ref)
+                .set("start", start)
+                .set("limit", 500);
+        String url = template.expand();
+        String response = getRequest(url);
         Map<String,Object> content = JsonParser.mapper.readValue(response, new TypeReference<Map<String,Object>>(){});
         Map page = (Map) content.get("children");
         List<Map> values = (List<Map>) page.get("values");
         collectFileAndDirectories(parent, path, values, files);
         while (!(boolean)page.get("isLastPage")){
-            response = getRequest(url + String.format("&start=%s&limit=%s", start, 500));
+            start += (int) content.get("size");
```

Comment:
We were never counting up the pages 😕 
Here we are 
https://github.com/casz/bitbucket-branch-source-plugin/blob/0f6a1060a67598df896406b11d15587c0541bf4e/src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java#L725-L729

However not here
https://github.com/casz/bitbucket-branch-source-plugin/blob/0f6a1060a67598df896406b11d15587c0541bf4e/src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java#L691-L695

Reply:
Can you create a JIRA and assign yourself (or let me know the JIRA and i’ll Assign it to you)

Good catch

Reply:
Jira issue created: https://issues.jenkins-ci.org/browse/JENKINS-48279